### PR TITLE
Django Upgrade to 2.2 - BOM-1011

### PR DESCRIPTION
**Pull request by mraarif**

### CODE REVIEW

**Comment by mraarif** [comment]
Build failure is due to the `quality` failure, I've tried changing the configuration for `pylint`, and looking at the build history in `travis`, it seems like only `pep8` was executing previously

**Comment by mraarif** [comment]
@jmbowman  in `tavis.yml` under `deploy` section it's targetting `python3.6` and `django1.11`, should we update that to `django2.2`

File: setup.py [hunk 1]
```diff
@@ -36,6 +37,7 @@
     license='AGPL',
     packages=['django_sites_extensions'],
     install_requires=[
-        'django>=1.8,<2.0',
+        'django>=1.8,<2.0;python_version<"3"',
+        'django>=2.2,<2.3;python_version>"3"'
```
[the remark this answers is not in the document]
Comment (mraarif) [reply setup.py]: so for `python3`, we won't be specifying any of the LTS versions i.e `1.11` or `2.2` strictly?

which one of the follwing requirement should be the part of `setup.py`?
```
'django>=1.11,<2.1;python_version>"3"',
 'django>=2.2,<2.3;python_version>"3"'
```
**or**
```
'django>=1.11,<2.1;python_version>"3"'
```

[the remark this answers is not in the document]
Comment (jmbowman) [reply setup.py]: I think we want something like this:

```
        'django>=1.11,<2.0 ; python_version < "3"',
        'django>=1.11 ; python_version > "3"',
```

Compared to what has been there, this:

* Drops support for Django versions below 1.11
* Still prevents installation of Django 2.0+ under Python 2 where those versions don't work
* Allows any Django versions still supported upstream under Python 3

**Pull request merged**